compiler/src/dotty/tools/dotc/transform/MoveStatics.scala

```diff
@@ -64,12 +90,24 @@ class MoveStatics extends MiniPhase with SymTransformer {
         else {
           val moduleTmpl = module.rhs.asInstanceOf[Template]
           val companionTmpl = companion.rhs.asInstanceOf[Template]
-          val (staticDefs, remainingDefs) = moduleTmpl.body.partition {
-            case memberDef: MemberDef => memberDef.symbol.isScalaStatic
-            case _ => false
+
+          val staticDefs = mutable.ListBuffer[Tree]()
+          val staticTiedDefs = mutable.ListBuffer[Tree]()
+          val localStaticDefs = mutable.ListBuffer[ValDef]()
+          val remainingDefs = mutable.ListBuffer[Tree]()
+
+          moduleTmpl.body.foreach {
+            case valDef: ValDef if valDef.symbol.isMethodLookupForLazy && valDef.symbol.isScalaStatic =>
+              localStaticDefs.addOne(valDef)
+            case memberDef: MemberDef if memberDef.symbol.isScalaStatic =>
+              if memberDef.symbol.isVarHandleForLazy then
+                staticTiedDefs.addOne(memberDef)
```

Comment:
It looks like this could be
```suggestion
                remainingDefs.addOne(memberDef)
```
which would remove the need for `staticTiedDefs` altogether.